vignettes/backtesting.Rmd

```diff
@@ -0,0 +1,460 @@
+
+The `arx_forecaster()` function wraps the autoregressive forecaster we need and
+comes with sensible defaults:
+
+- we specify the predicted outcome to be the percentage of doctor's visits with
+  CLI (`percent_cli`),
+- the predictors are the 7-day average of the case rate and the percentage of
+  doctor's visits with CLI (`percent_cli`),
+- we use a linear regression model as the engine,
+- the autoregressive features assume lags of 0, 7, and 14 days,
+- we forecast 7 days ahead.
+
+All these default settings and more can be seen by calling `arx_args_list()`:
+
+```{r}
+arx_args_list()
+```
+
+These can be modified as needed, by sending your desired arguments into
+`arx_forecaster(args_list = arx_args_list())`. For now we will use the defaults.
+
+__Note__: Unlike in the previous vignette, we will not train and forecast each
+geo location indivudally. Instead, we will use a __geo-pooled approach__, where
+we train the model on data from all states and territories combined. This is
+because the data is quite similar across states, and pooling the data can help
+improve the accuracy of the forecasts, while also reducing the susceptibility of
+the model to noise. In the interest of computational speed, we only use the 6
+state dataset here, but the full archive can be used in the same way and has
+performed well in the past. Implementation-wise, geo-pooling is achieved by
+simply dropping the `group_by(geo_value)` prior to `epix_slide()`.
+
+Let's use the `epix_as_of()` method to generate a snapshot of the archive at the
+last date, and then run the forecaster.
+
+```{r}
+# The .versions argument selects only the last version in the archive and
+# produces a forecast only on that date.
+forecasts <- archive_cases_dv_subset %>%
+  epix_slide(
+    ~ arx_forecaster(
+      .x,
+      outcome = "percent_cli",
+      predictors = c("case_rate_7d_av", "percent_cli")
+    )$predictions,
+    .versions = archive_cases_dv_subset$versions_end
+  )
+# Join the forecasts with with the latest data at the time of the forecast to
+# compare. Since `percent_cli` data has a few days of lag, we use `tidyr::fill` to
+# fill the missing values with the last observed value.
+forecasts %>%
+  inner_join(
+    archive_cases_dv_subset %>%
+      epix_as_of(archive_cases_dv_subset$versions_end) %>%
+      group_by(geo_value) %>%
+      tidyr::fill(percent_cli),
+    by = c("geo_value", "forecast_date" = "time_value")
+  ) %>%
+  select(geo_value, forecast_date, .pred, .pred_distn, percent_cli)
+```
+
+The resulting epi_df now contains two new columns: `.pred` and `.pred_distn`,
+corresponding to the point forecast (median) and the quantile distribution
+containing our requested quantile forecasts (in this case, 0.05 and 0.95)
+respectively. The forecasts look reasonable and in line with the data. The point
+forecast is close to the last observed value (joined in the table above as
+`percent_cli`), and the uncertainty bands are wide enough to capture the
```

Comment:
How are you evaluating the quality of the bands?

Reply:
Just whether the values are close, it's just meant to be a simple sanity check.

Reply:
The uncertainty bands aren't visible.

Reply:
Fixing that right now